core/src/main/java/org/infinispan/manager/DefaultCacheManager.java

```diff
@@ -134,7 +136,8 @@
    private final CacheContainerStats stats;
    private final ConfigurationManager configurationManager;
 
-   @GuardedBy("this")
+   private final ReadWriteLock stoppingLock = new ReentrantReadWriteLock();
+   @GuardedBy("stoppingLock")
    private boolean stopping;
```

Comment:
How about using [Lock#tryLock](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Lock.html#tryLock(long,%20java.util.concurrent.TimeUnit) method? If this method returns false we know we have to throw and exception here. WDYT?